plugins/vppv2/stnplugin/descriptor/stn.go

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate descriptor-adapter --descriptor-name STN  --value-type *stn.Rule --import "../model/stn" --output-dir "descriptor"
+
+package descriptor
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-errors/errors"
+	"github.com/gogo/protobuf/proto"
+	"github.com/ligato/cn-infra/logging"
+	scheduler "github.com/ligato/vpp-agent/plugins/kvscheduler/api"
+	ifDescriptor "github.com/ligato/vpp-agent/plugins/vppv2/ifplugin/descriptor"
+	"github.com/ligato/vpp-agent/plugins/vppv2/model/interfaces"
+	"github.com/ligato/vpp-agent/plugins/vppv2/model/stn"
+	"github.com/ligato/vpp-agent/plugins/vppv2/stnplugin/descriptor/adapter"
+	"github.com/ligato/vpp-agent/plugins/vppv2/stnplugin/vppcalls"
+)
+
+const (
+	// STNDescriptorName is the name of the descriptor for VPP STN rules
+	STNDescriptorName = "vpp-stn-rules"
+
+	// dependency labels
+	stnInterfaceDep = "stn-interface"
```

Comment:
Dependency labels usually need description that describe what we need the object to be, e.g. exist, have IP, be UP etc. So this should be: `stn-interface-exists`.